### PR TITLE
use latest feed price for wallet value

### LINKED ISSUE #2205: Eliminate 3 day averaging of estimated steem account value in wallet view
Motivation: whenever there's an increase in the price of Steem, it's likely to attract new buyers. But after they buy, they experience buyer's remorse because condenser reports a 3 day average of the price which is naturally lower than the actual price. Please replace this with a more timely value estimate, or remove it entirely. I strongly favor using a more accurate price versus removal, by the way.

### PR DESCRIPTION
**Pull request by originated**
Fixes #2205 - (update) I'm seeing a ci/circleci: run-blackboxtest error looking at it.

### CODE REVIEW

**Comment by originated** [comment]
getRecentPrice function updated to utilize last element looks right.  Something seems off with Redux getting the new value in the wallet

File: src/app/components/modules/UserWallet.jsx [hunk 1]
```diff
@@ -26,6 +26,7 @@ import {
 } from 'app/client_config';
 import * as transactionActions from 'app/redux/TransactionReducer';
 import * as globalActions from 'app/redux/GlobalReducer';
+import * as userActions from 'app/redux/UserReducer';
```
Comment (roadscape) [inline src/app/components/modules/UserWallet.jsx]: looks unused

File: src/app/components/modules/UserWallet.jsx [hunk 2]
```diff
@@ -765,11 +766,17 @@ export default connect(
...
     (state, ownProps) => {
         let price_per_steem = undefined;
-        const feed_price = state.global.get('feed_price');
-        if (feed_price && feed_price.has('base') && feed_price.has('quote')) {
-            const { base, quote } = feed_price.toJS();
-            if (/ SBD$/.test(base) && / STEEM$/.test(quote))
-                price_per_steem = parseFloat(base.split(' ')[0]);
+        let steem_api_price = state.user.get('steem_api_price');
+        if (steem_api_price) {
+            price_per_steem = steem_api_price;
```
Comment (roadscape) [inline src/app/components/modules/UserWallet.jsx]: probably needs to do base/quote here

File: src/app/redux/UserReducer.js [hunk 3]
```diff
@@ -68,6 +69,9 @@ export default function reducer(state = defaultState, action) {
             });
         }
 
+        case SET_PRICE_FEED:
+            return state.set('steemApiPrice', payload);
```
Comment (roadscape) [inline src/app/redux/UserReducer.js]: typo? in other places this is `steem_api_price`

File: src/app/redux/UserSaga.js [hunk 4]
```diff
@@ -67,6 +68,11 @@ export function* watchRemoveHighSecurityKeys() {
     yield* takeLatest('@@router/LOCATION_CHANGE', removeHighSecurityKeys);
 }
 
+function* getRecentPrice() {
+    const feed = yield call([api, api.getFeedHistoryAysnc]);
```
Comment (roadscape) [inline src/app/redux/UserSaga.js]: typo: `Aysnc`

File: src/app/redux/UserSaga.js [hunk 5]
```diff
@@ -67,6 +68,11 @@ export function* watchRemoveHighSecurityKeys() {
...
+    if (feed) yield put(userActions.setPriceFeed({price: feed[feed.length - 1]}));
```
Comment (roadscape) [inline src/app/redux/UserSaga.js]: `getFeedHistoryAsync` returns an object with the feed under the `price_history` key

**Comment by roadscape** [comment]
http://localhost:8080/@test25/transfers
0.509 STEEM POWER
dev: est value = $1.22 (~2.40/STEEM)
prod: est value = $1.27 (~2.50/STEEM)

Seems to be working now. 2.50 is the rolling median, and ~2.40 is the last median.

File: src/app/redux/UserSaga.js [hunk 6]
```diff
@@ -67,6 +68,15 @@ export function* watchRemoveHighSecurityKeys() {
     yield* takeLatest('@@router/LOCATION_CHANGE', removeHighSecurityKeys);
 }
 
+function* getLatestFeedPrice() {
+    const history = yield call([api, api.getFeedHistoryAsync]);
```
Comment (gl2748) [inline src/app/redux/UserSaga.js]: I think putting this in a try catch would be nice ...

File: src/app/redux/UserSaga.js [hunk 7]
```diff
@@ -67,6 +68,19 @@ export function* watchRemoveHighSecurityKeys() {
     yield* takeLatest('@@router/LOCATION_CHANGE', removeHighSecurityKeys);
 }
 
+function* getLatestFeedPrice() {
+    try {
+        const history = yield call([api, api.getFeedHistoryAsync]);
+        if (history) {
```
Comment (gl2748) [inline src/app/redux/UserSaga.js]: We don't need this if statement anymore, because if the `yield call` fails, the catch will get it...

Comment (roadscape) [reply src/app/redux/UserSaga.js]: Is it possible `yield call` can return blank without throwing an exception?

Comment (gl2748) [reply src/app/redux/UserSaga.js]: Cannot tell - where is this endpoint documented?

Comment (roadscape) [reply src/app/redux/UserSaga.js]: I'm not sure it's documented anywhere; https://github.com/steemit/steem/blob/bb8ca476f33c08434274f2d76333fcb74cfa58b8/libraries/plugins/apis/database_api/database_api.cpp#L158 is where the API is defined and https://github.com/steemit/steem/blob/62b48877d9f731c3fe00ef818e3324a0a3de3e63/libraries/chain/database.cpp#L3082 is where the structure is maintained

**Comment by roadscape** [comment]
Rebased on `master`.

**Pull request merged**